src/NHSD.BuyingCatalogue.Identity.Api/Services/LoginService.cs

```diff
@@ -23,65 +20,68 @@ internal sealed class LoginService : ILoginService, IDisposable
         }
 
         public void Dispose()
         {
             _userManager?.Dispose();
         }
 
-        public async Task<Result<SignInResult>> SignInAsync(string username, string password, Uri returnUrl)
+        public async Task<Result<SignInResponse>> SignInAsync(string username, string password, Uri returnUrl)
         {
             if (string.IsNullOrWhiteSpace(username) || string.IsNullOrWhiteSpace(password))
             {
-                errors.Add(LoginUserErrors.UserNameOrPasswordIncorrect());
-                return Result.Failure<SignInResult>(errors);
+                return Result.Failure<SignInResponse>(LoginUserErrors.UserNameOrPasswordIncorrect());
             }
 
-            var context = await _interaction.GetAuthorizationContextAsync(returnUrl?.ToString());
-
             var user = await _userManager.FindByNameAsync(username);
             if (user is null)
             {
-                errors.Add(LoginUserErrors.UserNameOrPasswordIncorrect());
-                return Result.Failure<SignInResult>(errors);
+                return Result.Failure<SignInResponse>(LoginUserErrors.UserNameOrPasswordIncorrect());
+            }
+
+            var context = await _interaction.GetAuthorizationContextAsync(returnUrl?.ToString());
+            var isUserValidResult = await ValidateUserCredentialsAsync(user, password);
+            if (!isUserValidResult.IsSuccess)
+            {
+                await _eventService.RaiseAsync(new UserLoginFailureEvent(user.UserName, ValidateUserMessage, clientId: context?.ClientId));
+
+                return Result.Failure<SignInResponse>(isUserValidResult.Errors);
             }
 
-            var isValidResult = await ValidateUserAsync(user, password, context);
-            if (!isValidResult)
+            var isUserDisabledResult = ValidateIfUserDisabled(user);
+
+            if (!isUserDisabledResult.IsSuccess)
             {
-                return Result.Failure<SignInResult>(errors);
+                await _eventService.RaiseAsync(new UserLoginFailureEvent(user.UserName, DisabledEventMessage, clientId: context?.ClientId));
+
+                return Result.Failure<SignInResponse>(isUserDisabledResult.Errors);
             }
 
             var props = new AuthenticationProperties
             {
                 RedirectUri = context?.RedirectUri,
                 AllowRefresh = true,
-                ExpiresUtc = DateTimeOffset.UtcNow.AddMinutes(_cookieExpirationSettings.ExpireTimeSpan.TotalMinutes),
                 IsPersistent = false
             };
 
             await _signInManager.SignInAsync(user, props);
 
             await RaiseLoginSuccessAsync(username, context);
-
             // We can trust returnUrl if GetAuthorizationContextAsync returned non-null
-            return Result.Success(context == null ? new SignInResult(true) : new SignInResult(true, true));
+            return Result.Success(context == null ? new SignInResponse() : new SignInResponse(true));
```

Comment:
Could this line be simplified? e.g.

`return Result.Success(new SignInResponse(context is object));`